### PR TITLE
Database: Fix alembic offline mode migration logic #5488

### LINKED ISSUE #5488: Fix alembic offline mode migration logic
Motivation
----------
Currently, in order to make an offline schema upgrade with alembic, one needs to specify the URL of the database just for the purpose of detecting the DB dialect.
https://github.com/rucio/rucio/blob/47e93cfbe5887071c70de4ba815c1bbdddfac2ce/lib/rucio/db/sqla/migrate_repo/env.py#L56

Furthermore, the migration script does not consider any starting revision from the configuration, this will end up producing all revisions unless the user specifies `alembic upgrade <CURRENT_REVISION>:head --sql` in the CLI.

Modification
------------

- Make the DB url configuration optional, and assume a `dialect` configuration option if the URL is not given. It makes sense to give the option to do the offline schema upgrade without having access to the DB in the working environment.
- Take into account the `starting_rev` configuration, in order to produce the upgrades from that starting point.

### PR DESCRIPTION
**Pull request by rizart**
This PR is a proof of concept for issue #5488.

### CODE REVIEW

**Comment by bari12** [comment]
LGTM @mlassnig any comment?

File: etc/alembic_offline.ini.template [hunk 1]
```diff
@@ -0,0 +1,78 @@
+# Copyright 2014-2019 CERN for the benefit of the ATLAS collaboration.
```
Comment (bari12) [inline etc/alembic_offline.ini.template]: Cosmetic, but please change this header to the current one. :smile

**Review by bari12 — CHANGES_REQUESTED** [review]
This can go, please just change the header.

File: etc/alembic_offline.ini.template (hunk 1 above)
Comment (rizart) [reply etc/alembic_offline.ini.template]: Should I also remove the authors as in the new header style?

Comment (bari12) [reply etc/alembic_offline.ini.template]: Yes 😄

File: etc/alembic.ini.template [hunk 2]
```diff
@@ -1,4 +1,5 @@
-# Copyright 2014-2019 CERN for the benefit of the ATLAS collaboration.
+# -*- coding: utf-8 -*-
+# Copyright CERN since 2014
```
Comment (mlassnig) [inline etc/alembic.ini.template]: The unified header for everything is now:

`Copyright European Organization for Nuclear Research (CERN) since 2012`

**Review by mlassnig — DISMISSED** [review]

File: etc/alembic.ini.template (hunk 2 above)
Comment (rizart) [reply etc/alembic.ini.template]: Thanks Mario, will fix! So, we put `2012` even for code/files written after that date?

Comment (mlassnig) [reply etc/alembic.ini.template]: Yes, because this covers the whole project.

**Review by bari12 — APPROVED** [review]

**Pull request merged**